docker/configure_workers_and_start.py

```diff
@@ -389,29 +554,219 @@ def generate_base_homeserver_config() -> None:
+        # name is actually requested.
+        requested_worker_name = ""
+        new_worker_type_string = worker_type_string
+        # First, check if a name is requested
+        if "=" in worker_type_string:
+            # Split on "=", remove extra whitespace from ends then make list
+            worker_type_split = split_and_strip_string(worker_type_string, "=")
+            if len(worker_type_split) > 2:
+                error(
+                    "To many worker names requested for a single worker, or to many "
+                    f"'='. Please fix: {worker_type_string}"
+                )
+            # if there was no name given, this will still be an empty string
+            requested_worker_name = worker_type_split[0]
+
+            # Check the last character of a requested name is not a number. This can
+            # cause an error that comes across during startup as an exception in
+            # 'startListening' and ends with 'Address already in use' for the port.
+            # This only seems to occur when requesting more than 10 of a given
+            # worker_type, otherwise it would be ok.
+            if requested_worker_name[-1].isdigit():
+                error(
+                    "Found a number at the end of the requested worker name: "
+                    f"{requested_worker_name}. This is not allowed as it will cause "
+                    "exceptions with 'Address already in use'. Recommend appending an "
+                    "underscore after the number if this what you really want to do."
+                )
+
+            # Reassign the worker_type string with no name on it.
+            new_worker_type_string = worker_type_split[1]
+
+        # At this point, we have:
+        #   requested_worker_name which might be an empty string
+        #   new_worker_type_string which might still be what it was when it came in
+
+        # Split the worker_type_string on "+", remove whitespace from ends then make
+        # the list a set so it's deduplicated. Hopefully no one tries to put 2
+        # pushers on the same worker(as it would consolidate into one).
+        worker_types_list = split_and_strip_string(new_worker_type_string, "+")
+        worker_types_set: Set[str] = set(worker_types_list)
+
+        # Shortcut this here, then it only has to survive intact(most of the time it
+        # will just pass right through)
+        worker_base_name = new_worker_type_string
+        if requested_worker_name:
+            worker_base_name = requested_worker_name
+            # It'll be useful to have this in the log in case it's a complex of many
+            # workers merged together. Note for Complement: it would only be seen in the
+            # logs for blueprint construction(which are not collected).
+            log(
+                f"Worker name request found: '{requested_worker_name}'"
+                f", for: {worker_types_set}"
+            )
+
+        else:
```

Comment:
I like the replace(*) options to strip the whitespace. Because it goes through the bash shell, do `\t`, `\n`, and `\r` stick around?

I guess we need to define a standard for "this is ok to mutate the name and move on, and this is a stop error". I don't preserve the order because of determinism(istics?), which I guess is one vote for mutating the name(that was not requested, in this case). Stripping the whitespace though I'm totally on board with.

Reply:
Actually, checking for spaces is done after this `for` loop after the de-dent just below, so it could probably be consolidated there instead of doing this part basically twice. What are the thoughts on `replace()`ing all those you suggest above and then running it through the regex?

Reply:
yeah, feel free to just strip out all the whitespace upfront if it's clearer

Reply:
I think for this, I would like to just use the `set` of worker_type(s). I think it would be better to just have this done with instead of replacing a bunch of potential nonsense. For context, this is if there is no name requested and we are going to use the worker_type(s) as the name. This data is already in a set(for deduplication), so it's order wasn't preserved. This way, there is no whitespace(as it would be gone already) and wouldn't have to run through the regex either. It would be checked for valid at a later time, when "checking for realism" as it's gathering data from `WORKERS_CONFIG`. Either that data exists or it doesn't, which is a stop-error. We need the sorted worker_type(s) anyways for the checklist to avoid one of those silly typos.

Having said that, above(when dealing with the requested name) I will `replace()` the spaces with underscores(as keeping things readable in logs is important, alternatively could just make spaces into a "camelCase" string because I have a sense of humor), but everything else is going through the regex. They can deal with their silly nonsense they are intentionally injecting if a `\t`, etc of all things ends up there.

Please tell me you are ok with this.

Reply:
Addressed in e2fecab
Tell me what you think about this

Reply:
yeah that's OK, I would also be fine with just rejecting whitespace but no strong opinions